.github/workflows/ci.yml

```diff
@@ -1,109 +1,52 @@
 name: CI
 
-on:
-  push:
-    branches:
-      - "*"
-  pull_request:
-    branches:
-      - "*"
-
-env:
-  CARGO_TERM_COLOR: always
-
-defaults:
-  run:
-    shell: bash
+on: [push, pull_request]
 
 jobs:
-  pr:
-    name: pull_request
-    if: "github.event_name == 'pull_request'"
-    strategy:
-      matrix:
-        os:
-          - ubuntu-20.04
-        toolchain:
-          - stable
-          - nightly
-    runs-on: ${{ matrix.os }}
-    needs: lint
+  check:
+    name: Run check
+    runs-on: ubuntu-latest
     steps:
-      - name: checkout the source code
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: ${{ matrix.toolchain }}
+          toolchain: nightly-2021-07-05
           override: true
-      - name: check
-        uses: actions-rs/cargo@v1
+      - run: rustup target add wasm32-unknown-unknown --toolchain nightly-2021-07-05
+      - uses: actions-rs/cargo@v1
         with:
           command: check
-      - name: test
-        if: ${{ matrix.toolchain == 'stable' }}
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-  master:
-    name: push
-    if: "github.event_name == 'push'"
-    strategy:
-      matrix:
-        os:
-          - ubuntu-20.04
-          - windows-2019
-          - macos-10.15
-        toolchain:
-          - stable
-          - beta
-          - nightly
-    runs-on: ${{ matrix.os }}
-    needs: lint
+
+  test:
+    name: Run unit test suite
+    runs-on: ubuntu-latest
     steps:
-      - name: checkout the source code
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v2
```

Comment:
There is quite a bit duplication between this and the previous section. I'm not particularly familiar with CI configs – can this be somehow unified?

Reply:
It's mostly to have these jobs running in parallel - but good point, I''ll see if there's any mechanism to unify these steps in actions, I'm less familiar with them than with CircleCI, but it should exists